File: Data Structures & Algorithms/max-water-container/submission-10.java
class Solution {
    public int maxArea(int[] heights) {

        int maxArea = 0, left = 0, right = heights.length - 1;

        while(left < right) {
            int currentArea = Integer.min(heights[left], heights[right]) * (right - left);
            maxArea = Integer.max(maxArea, currentArea);
            if(heights[left] <= heights[right]) {
                //maxArea = Integer.max(maxArea, currentArea);
                left++;
            }
            else {
                //maxArea = Integer.max(maxArea, currentArea);
                right--;
            }
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-10.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left, longest = 0, 0
        window = dict()

        for right in range(len(s)):
            window[s[right]] = 1 + window.get(s[right], 0)
            while (right - left + 1) - max(window.values()) > k:
                window[s[left]] -= 1
                left += 1
            longest = max(longest, right - left + 1)
        
        return longest